src/optimisation/Optimisation.jl

```diff
@@ -1,324 +1,327 @@
 module Optimisation
 
 using ..Turing
+using ..OptimisationCore: ModeResult, MLE, MAP, OptimLogDensity, OptimizationContext,
+    ModeEstimator, variable_names
+using NamedArrays: NamedArray
+using Optimization
+using OptimizationOptimJL: LBFGS, IPNewton
+using DocStringExtensions: TYPEDFIELDS
 using Bijectors
 using Random
-using SciMLBase: OptimizationFunction, OptimizationProblem, AbstractADType, NoAD
+using SciMLBase: OptimizationFunction, OptimizationProblem, solve
+using ADTypes: AbstractADType
```

Comment:
```suggestion
using SciMLBase: SciMLBase
using ADTypes: ADTypes
```

When using stuff from other modules, almost always prefer qualified references, i.e. we explicitly write out `SciMLBase.solve`.

When working in larger projects like Turing.jl, it's always difficult to know exactly what is from where, so we should try to make this as easy as possible to tell for the reader.

(Note that this is unfortunately not a pattern we've followed well enough up until now)

Reply:
Nice, very happy to conform to that, would actually be my preference as well. The Julia convention of just doing `using X; using Y` and leaving users mystified as to which package some later-referenced thing comes from baffles me.

Reply:
Done.